src/components/IDPage/IDPage.js

```diff
@@ -1,15 +1,26 @@
 import './IDPage.css';
 import MissionArea from '../MissionArea/MissionArea.js';
 import StoryArea from '../StoryArea/StoryArea.js';
+import Identity from '../Identity/Identity';
+import { useSelector } from 'react-redux';
+
+const IDPage = (props) => {
+  const {idNumber} = props
+  let currentIdentity = useSelector(state => state.identities.find(identity => {
+    console.log(identity.id)
+    return identity.id === parseInt(idNumber)} ))
+  let currentCountry = useSelector(state => state.countries.find(country => country.name === currentIdentity.country))
+ if(!currentCountry){
+   console.log(currentIdentity)
+   return <h1>Loading!</h1>
+ }
 
-const IDPage = () => {
   return (
     <section>
-      <h2>Name's Identity</h2>
       <h3>Your Backstory</h3>
-      <StoryArea />
+      <StoryArea currentIdentity = {currentIdentity}  currentCountry ={currentCountry}/>
       <h3>Your Mission</h3>
-      <MissionArea />
+      <MissionArea currentCountry ={currentCountry}/>
```

Comment:
I'm interested in the other way using state management and Redux, but this looks good so far. I'm glad you were able to use the data for the mission and story

Reply:
Yes, this works fine and implements react well. I think that knowing what the alternative is with redux principles is good, and either implementation is fine.